worker_pool.go

```
@@ -0,0 +1,133 @@
+				return
+			case <-ticker.C:
+				targetIdleWorkers := w.numMinIdleWorkers()
+
+				if w.pool.Len() <= targetIdleWorkers {
+					continue
+				}
+
+				nodes := w.pool.NodeSlice()
+				for _, node := range nodes[targetIdleWorkers:] {
+					if node.Value.GetLastUsed().Add(interval).Before(time.Now()) &&
+						!(node.Next() == nil && node.Prev() == nil) {
```

golangci-lint (2.12.1)

[error] QF1001: could apply De Morgan's law

(staticcheck)